File: src/Turis.DataAccessLayer/Entities/EventLog.cs
using Turis.Authentication.Entities;
using Turis.DataAccessLayer.Entities.Base;

namespace Turis.DataAccessLayer.Entities;

public class EventLog : BaseEntity
{
    public Guid UserId { get; set; }
    public ApplicationUser User { get; set; }
    public DateTimeOffset TimeStamp { get; set; }
    public string EntityName { get; set; }
    public string EntityKey { get; set; }
    public string EventName { get; set; }
    public string AdditionalInfo { get; set; }
}

File: src/Client/src/app/shared/event-log.ts
import { User } from 'app/core/user/user.types';
import { BaseSearchParameters } from './types/shared.types';

export class EventLog {
    id: string;
    user: User;
    date: Date;
    timeStamp: Date;
    eventName: string;
    additionalInfo: string;
}

export class EventLogSearchParameters extends BaseSearchParameters {
    entityName?: string;
    entityKey?: string;
    pattern?: string;
}
